bridgesample/src/main/java/com/livefront/bridgesample/scenario/fragment/LargeDataFragment.kt

```diff
@@ -31,6 +31,22 @@ class LargeDataFragment : BridgeBaseFragment() {
             setHeaderText(R.string.large_data_header)
             generatedBitmap = savedBitmap
             onBitmapGeneratedListener = { savedBitmap = it }
+            if (getArguments(this@LargeDataFragment).infiniteBackstack) {
+                onNavigateButtonClickListener = {
+                    this@LargeDataFragment.activity!!.supportFragmentManager
+                        .beginTransaction()
+                        .replace(
+                            (this@LargeDataFragment.view!!.parent as ViewGroup).id, newInstance(
+                                LargeDataArguments(
+                                    getArguments(this@LargeDataFragment).shouldClearOnDestroy,
+                                    true
+                                )
+                            )
+                        )
+                        .addToBackStack(null)
+                        .commit()
```

Comment:
Can we abstract this navigation logic a bit? We could make a `NavigationManager` class with a `navigateTo` method (or something like that). And then we have a `NavigationManagerProvider` interface that any relevant Activities implement to get access to the `NavigationManager`?

Reply:
I added `FragmentNavigationManager`, with a `navigateTo` method that takes a fragment to navigate to and a boolean for whether or not to add the fragment to the backstack. Now instead of the `(NonBridge)LargeDataFragment` assuming it is replacing the fragment in the same container, the activity it's contained in that implements `FragmentNavigationManager` can control it.